doc/modules/model_evaluation.rst

```diff
@@ -2023,6 +2051,15 @@ where :math:`\bar{y} = \frac{1}{n} \sum_{i=1}^{n} y_i` and :math:`\sum_{i=1}^{n}
 Note that :func:`r2_score` calculates unadjusted R² without correcting for
 bias in sample variance of y.
 
+In the particular case when the true y is constant, the R^2 score is not
+finite: it is either ``NaN`` (perfect predictions) or ``-Inf`` (imperfect
+predictions). Such non-finite scores may prevent correct model optimization
+such as grid-search cross-validation to be performed correctly. For this reason
+the default behaviour of :func:`r2_score` is to replace them with 1.0 (perfect
+predictions) or 0.0 (imperfect predictions). You can set the
+``fix_when_y_true_is_constant`` parameter to ``False`` to prevent this fix to
+happen and fallback on the original R².
```

Comment:
Not taking the parameter renaming suggested above into account, can this be simplified?
```suggestion
predictions) or 0.0 (imperfect predictions). If ``fix_when_y_true_is_constant``
is set to ``False``, this score fallbacks on the original R² definition.
```